meetings/2022-05-16.md

```diff
@@ -0,0 +1,95 @@
+
+* [Alternative Flows](https://github.com/solid/solid-oidc/blob/main/alternative-flows.md)
+
+
+## Minutes
+
+### Issues
+
+* AC: Is there any conversation we need to have about the batch of new issues?
+
+### Open PRs
+
+* AC: The only PR is something that has been open for long time, we already discussed it before.
+
+### Alternative Flows
+
+https://github.com/solid/solid-oidc/blob/main/alternative-flows.md
+
+* EP: Credit to Laurens for contributing extensively. The intention is to support clients acting on their own behalf, where there is no distinction between the Client and the RO. What WebID would be attributed to the client/RO then? Also, question abbout the clinet access control defaulting to closed.
+* AC: Note that the WebID and the Client ID are being conflated in this case. How can we support this? One idea would be: it you have an existing Client ID document, and augmenting (potentially using a secondary JSON-LD context). You could use this context to reference key material. The server could generate a nonce as a challenge for the client to demonstrate possession of a private portion of the public key material available in their client ID/WebID document.
+* EP: I think having apps acting on their own behalf could be an antipattern. There exists a distinction between clients and social agents which is described in the interoperability panel. An example is provided in the alternative flow document.
+* AC: What you describe is a very typical use case (app acting on behalf of a social agent) can be implemented using offline_access scope and refresh tokens. App acting on its own behalf is a different use case which isn't covered by Solid-OIDC currently. 
+* NC: When you receive the refresh token you set timeout to keep access token refreshed as long as the session is active.
+* EP: So refresh tokens itself don't have expiration?
+* NC: Token response includes access token and refresh token which are opaque to the client and information about expiration for the access token.
+* NC: When you restore persisted session, the refresh token is used to get new access token.
+* EP: I see there is assumption that refresh tokens are long lasting, weeks not hours or days.
+* AC: Using refresh tokens also often relies on the assumtion that some persistent storage layer is available.
+* EP: There are proposed solutions in `alternative-flows.md`. In order to implement them and support the discussed use case, should the Solid-OIDC spec be extended, or should new specs altogether be proposed?
+* AC: A difficulty we'll be facing is that device code flow/client credentials flow are OAuth2.0 contructs, while the ID token is an OIDC construct, defined at a different layer. We'll want to keep these concerns separate. Writing our own services makes that possible, but it will limit adoption of third-party software (e.g. CSS has faced issues implementing the client credential flow). Defining these will need to happen at the right level of abstraction.
```

Comment:
```suggestion
* AC: A difficulty we'll be facing is that device code flow and client credential flow are OAuth2.0 constructs, while the ID token is an OIDC construct, defined at a different layer. We'll want to keep these concerns separate. Writing our own services makes that possible, but it will limit adoption of third-party software (e.g., CSS has faced issues implementing the client credential flow). Defining these will need to happen at the right level of abstraction.
```